src/Xamarin.Android.Build.Tasks/Tasks/BuildApk.cs

```diff
@@ -380,6 +453,14 @@ CompressionMethod GetCompressionMethod (string fileName)
 		void AddNativeLibraryToArchive (ZipArchiveEx apk, string abi, string filesystemPath, string inArchiveFileName)
 		{
 			string archivePath = $"lib/{abi}/{inArchiveFileName}";
+			existingEntries.Remove (archivePath);
+			if (apk.Archive.ContainsEntry (archivePath)) {
+				DateTime lastWrite = File.GetLastWriteTimeUtc (filesystemPath);
+				if (lastWrite <= apk.Archive.ReadEntry (archivePath).ModificationTime) {
+					Log.LogDebugMessage ($"Skipping {filesystemPath} (APK path: {archivePath}) as it is up to date.");
+					return;
+				}
+			}
```

Comment:
There are a few places code exactly like this. I think we should put it on `ZipArchiveEx` so it isn't duplicated?

We could also add some unit tests for `ZipArchiveEx` to be sure it works as expected.